src/containers/PipelineResourcesDropdown/PipelineResourcesDropdown.js

```diff
@@ -55,9 +58,17 @@ PipelineResourcesDropdown.defaultProps = {
 
 function mapStateToProps(state, ownProps) {
   const namespace = ownProps.namespace || getSelectedNamespace(state);
+  const { type } = ownProps;
+  const pipelineResourcesReducer = (acc, pipelineResource) => {
+    if (!type || type === pipelineResource.spec.type) {
+      acc.push(pipelineResource.metadata.name);
+    }
+    return acc;
+  };
   return {
-    items: getPipelineResources(state, { namespace }).map(
-      pipelineResource => pipelineResource.metadata.name
+    items: getPipelineResources(state, { namespace }).reduce(
```

Comment:
We could use `filter` here rather than `reduce`, something like:
`getPipelineResource(...).filter(resource => (!type || type === resource.spec.type))`

Reply:
Ok, I'll have to do `filter` to `map` though to end with a string.